lib/board.rb

```diff
@@ -0,0 +1,46 @@
+class Board
+  attr_accessor :cells
+
+  @cells = []
+
+  def initialize
+    self.cells = Array.new(9, " ")
+  end
+
+  def reset!
+    self.cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{self.cells[0]} | #{self.cells[1]} | #{self.cells[2]} "
+    puts "-----------"
+    puts " #{self.cells[3]} | #{self.cells[4]} | #{self.cells[5]} "
+    puts "-----------"
+    puts " #{self.cells[6]} | #{self.cells[7]} | #{self.cells[8]} "
+  end
+
+  def position(user_input)
+    self.cells[user_input.to_i-1]
+  end
+
+  def full?
+    self.cells.all? {|cell| cell == "X" || cell == "O"}
+  end
+
+  def turn_count
+    self.cells.count("X") + self.cells.count("O")
+  end
+
+  def taken?(position)
+    self.cells[position.to_i-1] == "X" || self.cells[position.to_i-1] == "O"
+  end
+
+  def valid_move?(position)
+    !taken?(position) && position.to_i > 0 && position.to_i <= 9
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`position.between?(1,9)`